entity-service-client/src/main/java/org/hypertrace/entity/data/service/client/EdsTypeAndIdAttributesCacheKey.java

```diff
@@ -0,0 +1,40 @@
+package org.hypertrace.entity.data.service.client;
+
+import java.util.Objects;
+import org.hypertrace.entity.data.service.v1.ByTypeAndIdentifyingAttributes;
+
+public class EdsTypeAndIdAttributesCacheKey {
+
+  public final String tenantId;
+  public final ByTypeAndIdentifyingAttributes byTypeAndIdentifyingAttributes;
+
+  public EdsTypeAndIdAttributesCacheKey(String tenantId, ByTypeAndIdentifyingAttributes
+      byTypeAndIdentifyingAttributes) {
+    this.tenantId = tenantId;
+    this.byTypeAndIdentifyingAttributes = byTypeAndIdentifyingAttributes;
+  }
+
+  @Override
+  public String toString() {
+    return String.format("__%s__%s", this.tenantId, this.byTypeAndIdentifyingAttributes);
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(tenantId, byTypeAndIdentifyingAttributes);
```

Comment:
This triggers the `hashCode` calculation on the `byTypeAndIdentifyingAttributes` object internally. I've not fully seen the proto auto-generated code for that but did you think about performance impact due to that? Is it better to handle this on our own rather?

Reply:
Might be tricky to maintain if we have a custom implementation

Reply:
Scanned the internal Impl of generated proto's equals and hashCode. I think we should be fine.